hoomd_organics/library/forcefields.py

```diff
@@ -182,3 +183,284 @@ def _create_forcefield(self):
+    ----------
+    pairs: dict, optional, default None
+    bonds: dict, optional, default None
+    angles: dict, optional, default None
+    dihedrals: dict, optional, default None
+
+    """
+
+    def __init__(
+        self,
+        pairs=None,
+        bonds=None,
+        angles=None,
+        dihedrals=None,
+        r_min=None,
+        r_cut=None,
+        exclusions=["bond", "1-3"],
+        nlist_buffer=0.40,
+    ):
+        self.pairs = pairs
+        self.bonds = bonds
+        self.angles = angles
+        self.dihedrals = dihedrals
+        self.r_min = r_min
+        self.r_cut = r_cut
+        self.exclusions = exclusions
+        self.nlist_buffer = nlist_buffer
+        self.bond_width, self.angle_width, self.dih_width = self._check_widths()
+        hoomd_forces = self._create_forcefield()
+        super(TableForcefield, self).__init__(hoomd_forces)
+
+    @classmethod
+    def from_files(
+        cls,
+        pairs=None,
+        bonds=None,
+        angles=None,
+        dihedrals=None,
+        exclusions=["bond", "1-3"],
+        nlist_buffer=0.40,
+        **kwargs,
+    ):
+        """Create a table forefield from files.
+
+        Parameters
+        ----------
+        pairs: dict, optional, default None
+            Dictionary with keys of pair type and keys of file path
+        bonds: dict, optional, default None
+            Dictionary with keys of bond type and keys of file path
+        angles: dict, optional, default None
+            Dictionary with keys of angle type and keys of file path
+        dihedrals: dict, optional, default None
+            Dictionary with keys of dihedral type and keys of file path
+        ``**kwargs`` : keyword arguments
+            Key word arguments passed to `numpy.genfromtxt` or `numpy.load`
+
+        Notes
+        -----
+        The parameters must use a `{"type": "file_path"}` mapping.
+
+        Following HOOMD conventions, pair types must be given as a `tuple`
+        of particles types while bonds, angles and dihedrals
+        are given as a `str` of particle types separated by dashes.
+
+        Example
+        -------
+        .. code-block:: python
+
+            table_forcefield = TableForcefield.from_files(
+                pairs = {
+                    ("A", "A"): "A_pairs.txt"
+                    ("B", "B"): "B_pairs.txt"
+                    ("A", "B"): "AB_pairs.txt"
+                },
+                bonds = {"A-A": "A_bonds.txt", "B-B": "B_bonds.txt"},
+                angles = {"A-A-A": "A_angles.txt", "B-B-B": "B_angles.txt"},
+            )
+
+        Warning
+        -------
+        It is assumed that the structure of the files are:
+            * Column 1: Independent variable (e.g. distance, length, angle)
+            * Column 2: Energy
+            * Column 3: Force
+
+        """
+
+        def _load_file(file, **kwargs):
+            """Call the correct numpy method."""
+            if file.split(".")[-1] in ["txt", "csv"]:
```

Comment:
Also probably need some validations here on the file name and file path. 
For example, raising error if the file is not txt or csv or numpy or doing a `os.path.exists` check before loading.

Reply:
Good idea.